### PR TITLE
chore: replace black with ruff as the formatter

### PR DESCRIPTION
**Pull request by wood-push-melon**
This pull request:

- replaces black with ruff as the formatter
- fixes the lint and format issues with ruff
- keeps the isort to keep the original behaviors for import sorting
- puts the codespell configuration to the pyproject.toml file to be consistent with other tools

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -12,29 +12,74 @@ show_missing = true
...
+    "tests/**/*.py",
+    "lib/charms/identity_platform_login_ui_operator/**/*.py"
+]
+extend-exclude = ["__pycache__", "*.egg_info"]
+target-version = "py38"
 
-[tool.isort]
-line_length = 99
-profile = "black"
+[too.ruff.lint]
```
Comment (natalian98) [inline pyproject.toml]:
```suggestion
[tool.ruff.lint]
```

File: pyproject.toml [hunk 2]
```diff
@@ -12,29 +12,74 @@ show_missing = true
...
+    "E501",
+    "N818"
+]
+per-file-ignores = {"tests/*" = ["D100","D101","D102","D103","D104"]}
 
-# Linting tools configuration
-[tool.flake8]
-max-line-length = 99
-max-doc-length = 99
+[too.ruff.lint.flake8-copyright]
```
Comment (natalian98) [inline pyproject.toml]:
```suggestion
[tool.ruff.lint.flake8-copyright]
```

File: pyproject.toml (hunk 1 above)
Comment (BarcoMasile) [inline pyproject.toml]: issue: typo
```suggestion
[tool.ruff.lint]
```

File: pyproject.toml (hunk 2 above)
Comment (BarcoMasile) [inline pyproject.toml]: issue: typo
```suggestion
[tool.ruff.lint.flake8-copyright]
```

**Review by BarcoMasile — CHANGES_REQUESTED** [review]
Typo in `pyproject.toml` that Natalia caught

**Review by nsklikas — APPROVED** [review]

**Review by natalian98 — APPROVED** [review]

**Review by BarcoMasile — APPROVED** [review]

**Pull request merged**